kratos/mpi/utilities/parallel_fill_communicator.h

```diff
@@ -350,17 +345,16 @@ class ParallelFillCommunicator
         }
 
         // Initialize arrays for all neighbour id lists on root process.
-        std::vector<unsigned> number_of_receive_neighbours;
+        std::vector<std::size_t> number_of_receive_neighbours;
```

Comment:
why do you sometimes use `unsigned int` and sometimes `size_t`?
Both are available in the DataCommunicator now right?

Reply:
The existing code used (and still uses) vectors of `unsigned int` for communication, but at some points we are just sending the output of `some_vector.size()`, which I kept as a `size_t` when possible to avoid creating a copy on the spot.